pkg/monitor/cluster/hiveregistrationstatus.go

```diff
@@ -0,0 +1,90 @@
+package cluster
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+
+	hive "github.com/Azure/ARO-RP/pkg/hiveregistrator"
+)
+
+type hiveRegistrationState struct {
+	isRegistered        bool
+	isUnreachable       bool
+	notRegisteredReason string
+	notReachableReason  string
+}
+
+func (mon *Monitor) emitHiveRegistrationStatus(ctx context.Context) error {
+	state, err := mon.hiveRegistrationState(ctx)
+	if err != nil {
+		return err
+	}
+
+	if !state.isRegistered {
+		mon.emitGauge("hive.registration.error", 1, map[string]string{
+			"reason": state.notRegisteredReason,
+		})
+	}
+
+	if state.isUnreachable {
+		mon.emitGauge("hive.registration.isUnreachable", 1, map[string]string{
+			"reason": state.notReachableReason,
+		})
+	}
+
+	return nil
+}
+
+func (mon *Monitor) hiveRegistrationState(ctx context.Context) (hiveRegistrationState, error) {
+	state := hiveRegistrationState{
+		isRegistered:        false,
+		isUnreachable:       false,
+		notRegisteredReason: "",
+		notReachableReason:  "",
+	}
+
+	if mon.oc.Properties.HiveClusterNamespace == "" {
+		state.notRegisteredReason = "no namespace in cluster document"
+		return state, nil
+	}
+
+	if mon.oc.Properties.HiveClusterdeploymentName == "" {
+		state.notRegisteredReason = "no clusterdeploymentname in cluster document"
+		return state, nil
+	}
+
+	hiveClients, err := hive.HiveClientsFromFileEnvVariable()
+	if err != nil {
+		return state, err
+	}
+
+	hr := hive.New(&hive.WorkloadCluster{
+		HiveClusterdeploymentName: mon.oc.Properties.HiveClusterdeploymentName,
+		HiveNamespaceName:         mon.oc.Properties.HiveClusterNamespace,
+	}, hiveClients)
+
+	isRegistered, reason, err := hr.IsRegistered(ctx)
+	if err != nil {
+		return state, err
+	}
+
+	if !isRegistered {
+		state.notRegisteredReason = reason
+		return state, nil
+	}
+
+	state.isRegistered = true
+
+	isConnected, reason, err := hr.IsConnected(ctx)
+	if err != nil {
+
+		return state, err
+	}
+	if !isConnected {
```

Comment:
per my suggestion we should then check for the true value of this bool
```suggestion
        if isConnected {
          state.Reachable = true
        }

Reply:
and then you'd need to properly set the notReachableReason in the false case